File: 02_Recursion,_Divide_&_Conquer/2447_Fractal_Asterisk_Pattern/Answer2447.ts
const n = +`${require('fs').readFileSync(0)}`

for(let y = 0; y < n; y++){
  let line = ''
  for(let x = 0; x < n; x++){
    let tmpX = x
    let tmpY = y
    let tmpS = '*'
    while(tmpX >= 1 && tmpY >= 1){
      if(tmpY  % 3 === 1 && tmpX % 3 === 1){
        tmpS = ' '
        break;
      }
      tmpX = tmpX / 3 | 0
      tmpY = tmpY / 3 | 0
    }
    line += tmpS
  }
  console.log(line)
}

File: 02_Recursion,_Divide_&_Conquer/2447_Fractal_Asterisk_Pattern/Answer2447.java
import java.util.Scanner;

public class Answer2447 {
  public static void main(String[] args) {
    int n = new Scanner(System.in).nextInt();

    for(int y = 0; y < n; y++){
      var line = new StringBuilder();
      for(int x = 0; x < n; x++){
        int tmpX = x,
            tmpY = y;
        var tmpS = "*";
        while(tmpX >= 1 && tmpY >= 1){
          if(tmpY  % 3 == 1 && tmpX % 3 == 1){
            tmpS = " ";
            break;
          }
          tmpX /= 3;
          tmpY /= 3;
        }
        line.append(tmpS);
      }
      System.out.println(line);
    }
  }
}
